src/java/main/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -303,15 +305,38 @@ public void loadData() throws IOException, InterruptedException {
 
     public void takeSnapshot(){
```

Comment:
@revans2 It's not easy to find `SyncRequestProcessor `, but we may use [reversed workaroud](https://github.com/apache/zookeeper/pull/180/files#diff-a8615898f7be77e9cfdb686e21c97b1eL123), let's `SyncRequestProcessor` use `zks.getLastProcessedZxid()-zks.getLastSnapshotZxid()` to trace snapshot